hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/StringUtils.java

```diff
@@ -1182,6 +1200,19 @@ public static String popFirstNonOption(List<String> args) {
     }
     return null;
   }
+  /**
+   * From a list of command-line arguments, ensure that all of the arguments
+   * have been used except a possible "--".
+   *
+   * @param args  List of arguments.
+   * @throws IllegalArgumentException if some arguments were not used
+   */
+  public static void ensureAllUsed(List<String> args) throws IllegalArgumentException {
+    if (!args.isEmpty() && !(args.size() == 1 && "--".equals(args.get(0)))) {
```

Comment:
I don't think we need some of the parenthesis.

Reply:
I find writing code that depends overly on knowing the precedence for non-math operators leads to trouble, so I'd prefer to leave them in.